pkg/convert/pprof/streaming/vt_function.go

```diff
@@ -0,0 +1,155 @@
+package streaming
+
+import (
+	"fmt"
+	"io"
+)
+
+// revive:disable-next-line:cognitive-complexity,cyclomatic necessary complexity
+func (m *function) UnmarshalVT(dAtA []byte) error {
```

Comment:
These unmarshal methods are very repetitive. Can we do anything with it?

Reply:
I don't think it is a problem for small structs like function and locations
For Profile I had 3 huge almost identical copies for 3 passes, I merged them to a single function with an "operation" parameter